plutarch-docs/Concepts/GenericProgramming.md

```diff
@@ -126,8 +126,8 @@ instance DerivePlutusType (MyType a b) where type DPTStrat _ = PlutusTypeScott
 -- If you instead want to use data encoding, you should derive 'PlutusType' and provide data strategy:
 
 data MyTypeD (a :: PType) (b :: PType) (s :: S)
-  = OneD (Term s (PDataRecord '[ "_0" ':= a ]))
-  | TwoD (Term s (PDataRecord '[ "_0" ':= b ]))
+  = OneD (Term s (PDataRecord '[ "_0" ' := a ]))
```

Comment:
wtf?

Reply:
Two possibilities:
- Fourmolu is misconfigured -> fix configuration
- You can't configure this -> remove leading `'`

Reply:
It's not fourmolu problem. GHC gives warning if `':=` for some reason. 

```
 The suffix use of a ':<>:' might be repurposed as special syntax
      by a future language extension.
    Suggested fix: Add whitespace around ':<>:'.
```

I don't think turning off ticking infix tyop is a good idea either

Reply:
That’s a GHC bug if you ask me

Reply:
Yeah, yet another weird behavior of GHC parser. It should really enforce this whitespace rule after infix ticks so like `_':<>:_` not `'_:<>:_`

Reply:
Just remove the tick though? Also perhaps report a GHC bug if you have time?

Reply:
Ticks are quite useful, I don't think we should remove them.

Reply:
I’m not working with Plutarch anymore so do whatever you fancy but it has been decided that Haskell will be a dependently typed language and GHC doesn’t even warn on missing ticks anymore, they’re merely an awkward way to tell the compiler whether you want the promoted constructor or the datatype in the presence of punning. I’d advocate for simply not using punning, wherever possible (also something where the Haskell ecosystem is decidedly headed btw)

Reply:
> it has been decided that Haskell will be a dependently typed language
Hell, this sounds great. 

> I’d advocate for simply not using punning, wherever possible (also something where the Haskell ecosystem is decidedly headed btw)
This would be ideal, but even prelude itself have a ton of types with punning. At current stage, I don't think we are quite there to remove them entirely. More likely in the future. 

Made issue: https://gitlab.haskell.org/ghc/ghc/-/issues/24208

Reply:
Revert change and disable warning or remove tick.